src/examples/button/button.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"machine"
+	"runtime"
+)
+
+func main() {
+	led := machine.GPIO{machine.LED}
+	led.Configure(machine.GPIOConfig{Mode: machine.GPIO_OUTPUT})
+
+	button := machine.GPIO{8}
```

Comment:
That's a hardcoded pin number, which I would really like to avoid. Sadly the Arduino doesn't have a button on the board (unlike PCA10040), so let's just pick one (8 is fine). It can be set in `machine_avr.go` as something like `machine.BUTTON`. It's important to configure it in the machine module, as the PCA10040 has different numbers (13-16).
Thinking about it, `machine_avr.go` is also the wrong place for pin numbers, it should be `machine_arduino.go` with proper build constraints. But that can be fixed separately.

Reply:
See https://github.com/aykevl/tinygo/pull/7/commits/0266b63d5661ff13b341c4ad85bbe7c00fc2acf1 for what I mean.

Reply:
Hmmm that commit certainly improves things. What I'd really like to see is a way to have the code compile on both systems without change, but using the appropriate pin numbers. I don't see an obvious way to do it so let's keep it as it is now.